### PR TITLE
[PATH] Adaptive improve helix entry

### PR DESCRIPTION
**Pull request by Schildkroet**
Added support for conical entry helix
image

### CODE REVIEW

File: src/Mod/Path/PathScripts/PathAdaptive.py [hunk 1]
```diff
@@ -88,6 +83,22 @@ def discretize(edge, flipDirection = False):
 
     return pts
 
+def CalculateCone(radius, alpha):
```
Comment (dubstar-04) [inline src/Mod/Path/PathScripts/PathAdaptive.py]: Is this function used?

File: src/Mod/Path/PathScripts/PathAdaptiveGui.py [hunk 2]
```diff
@@ -78,13 +78,22 @@ def getForm(self):
...
         # helix angle
         form.HelixAngle = QtGui.QDoubleSpinBox()
-        form.HelixAngle.setMinimum(0.1)
-        form.HelixAngle.setMaximum(90)
-        form.HelixAngle.setSingleStep(0.1)
-        form.HelixAngle.setValue(5)
+        form.HelixAngle.setMinimum(0.5)
+        form.HelixAngle.setMaximum(359)
+        form.HelixAngle.setSingleStep(1)
+        form.HelixAngle.setValue(180)
```
Comment (dubstar-04) [inline src/Mod/Path/PathScripts/PathAdaptiveGui.py]: These values are a bit confusing. I would expect the helix angle to be a value between 0.1 and 90 where 0.1 is a very shallow helix and 90 is a normal z move?

Comment (dubstar-04) [reply src/Mod/Path/PathScripts/PathAdaptiveGui.py]: HelixAngle

Comment (dubstar-04) [reply src/Mod/Path/PathScripts/PathAdaptiveGui.py]: https://en.wikipedia.org/wiki/Helix_angle

File: src/Mod/Path/PathScripts/PathAdaptive.py (hunk 1 above)
Comment (Schildkroet) [reply src/Mod/Path/PathScripts/PathAdaptive.py]: This function was already removed in last commit

File: src/Mod/Path/PathScripts/PathAdaptiveGui.py (hunk 2 above)
Comment (Schildkroet) [reply src/Mod/Path/PathScripts/PathAdaptiveGui.py]: I changed it to 360° for a full circle, but i can change it back to 90

**Comment by sliptonic** [comment]
I'm willing to merge this because it works as intended.  However I find the feature a bit confusing and potentially dangerous so I encourage additional work on it. 

I would rather see the logic merged into the helix feature because it could be useful for creating conical pockets. 

Here's why I think it's confusing: 
I don't understand what is gained by adding this kind of entry to the adaptive clearing.  Does it improve finish? speed? 

Here's why I think it's dangerous in present form: 

The user can specify a very large cone angle (~45 degrees),  This results in 100% tool engagement radially as the cutter starts to descend. That's normal and unavoidable.  
However, Since the cone is much larger than the tool, a 'stem'  is left behind in the center.  As the tool descends, the axial engagement on the tool increases with each step down because the tool is cutting more and more of the leftover stem.  
The result is that the load on the tool increases continually.  A user might monitor the cut for the first few minutes and conclude that it is cutting well, but the load will potentially increase to the point of failure.


image
image

If you want to proceed with merge, please let me know.  I'll go ahead but please document the feature fully on the wiki.

**Comment by Schildkroet** [comment]
I will do some improvements regarding the mentioned problems

**Comment by Thorhian** [comment]
> I'm willing to merge this because it works as intended. However I find the feature a bit confusing and potentially dangerous so I encourage additional work on it.
> 
> I would rather see the logic merged into the helix feature because it could be useful for creating conical pockets.
> 
> Here's why I think it's confusing:
> I don't understand what is gained by adding this kind of entry to the adaptive clearing. Does it improve finish? speed?
> 
> If you want to proceed with merge, please let me know. I'll go ahead but please document the feature fully on the wiki.

I actually use this in Fusion 360. It can help prevent chattering/rubbing against the stock in the entry helix. Finding a way to add this to the helix feature and use the helix op inside the adaptive tool op would be great IMO.

**Comment by Schildkroet** [comment]
I limited the angle to 6°
Since this feature only tries to ease the entry with the helix in adaptiv and doesn't produce any useful surface, it makes no sense to also use it with the standalone helix op

**Pull request merged**